### ISSUE TITLE
Suddenly the built in mongo connector stopped the source

### ISSUE THREAD

**Issue by prasathsekar** [apache/pulsar#22777]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Read release policy

- [X] I understand that unsupported versions don't get bug fixes. I will attempt to reproduce the issue on a supported version of Pulsar client and Pulsar broker.

### Version

3.2.3

### Minimal reproduce step

{
"configs": {
"mongoUri": "{{mongo_uri}}",
"database": "{{db_name}}",
"collection": "chub_agent",
"syncType": "incr_sync",
"batchSize": "2",
"batchTimeMs": "500"
}
}

### What did you expect to see?

Any changes in the collection will be sourced and pushed to the pulsar topic

### What did you see instead?

We used to source 10 collections. suddenly the connector stops the source from MongoDB. There is no error, it show running but no file is sourced.

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

**Comment by eddydecena** [comment]
> Going through the basic troubleshooting questions:
> 
> * Did you confirm that the DB credentials are still valid and you can connect to the DB using a different tool?

Yes, we have other applications consuming the database with the same credentials.

> * Are there any log files associated with the connector that you can share, e.g. K8s pod log, etc?

I checked the logs and I don't see any errors.

> * Have you tried stopping/restarting the connector?

Yes

> * Have you tried deleting/re-creating the connector?

Yes
